### ISSUE TITLE
Regression in key conversion in 0.9.7

### ISSUE THREAD

**Issue by EpicEric** [RustCrypto/RSA#463]
TL;DR Regression pertaining to 2edd47929f7298bddde8f682ded330dbaf9c43a6.

With the 0.9.7 release, this created breakage when converting from an `RsaKeypair` to a `SigningKey`. More specifically, in the context of [`ssh_key::Signer::try_sign`](https://github.com/RustCrypto/SSH/blob/023b3606546451798b41ae653e38d8017beae8e1/ssh-key/src/signature.rs#L666), which has caused tests with a dependency on `rsa = "0.9"` to fail in [`russh` due to this call](https://github.com/Eugeny/russh/blob/ff95994945db593115dd0d663f8c41ecba945b9f/russh-keys/src/agent/server.rs#L345). I've tried refreshing the key used for tests, but it still results in an error due to this change.

The only fix I found was to pin the dependency on `rsa` to version 0.9.6, as seen in https://github.com/Eugeny/russh/pull/400.

**Comment by dignifiedquire** [comment]
This sounds like you are trying to sign using an invalid key, or we have an underlying bug in key validation. Do you have a key that can reproduce the issue?

**Comment by EpicEric** [comment]
Here's a minimal reproduction using a PKCS8 encrypted key: https://github.com/EpicEric/rsa-regression-example

As it shows, `RsaPrivateKey::from_components()` succeeds, but converting to an `ssh_key::KeypairData` before trying to sign fails when it shouldn't.

**Comment by dignifiedquire** [comment]
Thanks for the repo, I just looked into it, and the failure happens in the key conversion, and not in actual signature generation here: 

https://github.com/RustCrypto/SSH/blob/ssh-key/v0.6.7/ssh-key/src/signature.rs#L668

```rust
let data = rsa::pkcs1v15::SigningKey::<Sha512>::try_from(self)?
```

I don't know yet if the key itself is invalid, or there is an issue in the conversion.

**Comment by EpicEric** [comment]
I've generated a new key with the following command, and the conversion error persists:

```shell
openssl genrsa -aes128 -passout pass:blabla 3072
```

**Comment by dignifiedquire** [comment]
What a labyring of traits..but I found the bug that was revealed by the validation: https://github.com/RustCrypto/SSH/blob/ssh-key/v0.6.7/ssh-key/src/private/rsa.rs#L201-L202

```rust
  rsa::BigUint::try_from(&key.private.p)?,
  rsa::BigUint::try_from(&key.private.p)?,
```

This is passing the `p` prime twice, instead of `p` and `q` creating an invalid key. Before this was left silently and invalid signatures were generated.
